### PR TITLE
tests: improve drag-and-drop coverage

### PR DESCRIPTION
**Pull request by Remexllee**
**Summary**

I fixed the unit test in drag-and-drop-test.js file

This is a refactoring of unit test

**Related Issues/PRs**

### CODE REVIEW

File: lighthouse-viewer/test/drag-and-drop-test.js [hunk 1]
```diff
@@ -17,8 +15,12 @@ testHelpers.setupJsDomGlobals();
 
 const DragAndDrop = require('../app/src/drag-and-drop.js');
 
-function assertUIReset() {
-  assert.ok(!document.querySelector('.drop_zone').classList.contains('dropping'));
+function createCustomEvent(type, key, value) {
```
Comment (patrickhulce) [inline lighthouse-viewer/test/drag-and-drop-test.js]: nit: WDYT about the caller passing `{key: value}` to make it a little clearer below

File: lighthouse-viewer/test/drag-and-drop-test.js [hunk 2]
```diff
@@ -29,31 +31,59 @@ describe('DragAndDrop', () => {
...
+    new DragAndDrop(mockCallback);
+
+    // create custom drop event with mock files in dataTransfer
+    document.dispatchEvent(createCustomEvent('drop', 'dataTransfer', {
+      files: ['mock file'],
+    }));
+    expect(mockCallback).toBeCalledWith('mock file');
+  });
+
+  it('document responds to drop event without file', () => {
```
Comment (patrickhulce) [inline lighthouse-viewer/test/drag-and-drop-test.js]:
```suggestion
  it('document ignores drop event without file', () => {
```

File: lighthouse-viewer/test/drag-and-drop-test.js [hunk 3]
```diff
@@ -17,8 +15,12 @@ testHelpers.setupJsDomGlobals();
 
 const DragAndDrop = require('../app/src/drag-and-drop.js');
 
-function assertUIReset() {
-  assert.ok(!document.querySelector('.drop_zone').classList.contains('dropping'));
+function createCustomEvent(type, key, value) {
+  const customDropEvent = new window.CustomEvent(type);
+  if (key && value) {
```
Comment (patrickhulce) [inline lighthouse-viewer/test/drag-and-drop-test.js]: right now it's a bit odd that the callers do `createCustomEvent('drag', 'dataTransfer', null)` when it seems like that doesn't really do anything, is this intentional?

File: lighthouse-viewer/test/drag-and-drop-test.js [hunk 4]
```diff
@@ -29,31 +31,59 @@ describe('DragAndDrop', () => {
...
+    expect(dragoverEvent.dataTransfer).toBeUndefined();
   });
 
-  it('document responds to drag and drop events', () => {
-    // eslint-disable-next-line no-unused-vars
-    const dragAndDrop = new DragAndDrop();
+  it('document responds to mouseleave event when not dragging', () => {
+    new DragAndDrop(jest.fn);
 
     document.dispatchEvent(new window.CustomEvent('mouseleave'));
```
Comment (patrickhulce) [inline lighthouse-viewer/test/drag-and-drop-test.js]: should we align on `createCustomEvent` while we're at it?

Comment (Remexllee) [reply lighthouse-viewer/test/drag-and-drop-test.js]: true, I will made the change align it

File: lighthouse-viewer/test/drag-and-drop-test.js (hunk 3 above)
Comment (connorjclark) [reply lighthouse-viewer/test/drag-and-drop-test.js]: I think removing this fn entirely would be better.

```
// create custom drop event with mock files in dataTransfer
const event = new CustomEvent('drop');
event.dataTransfer = {
  files: ['mock file'],
};
document.dispatchEvent(event);
```

Comment (patrickhulce) [reply lighthouse-viewer/test/drag-and-drop-test.js]: Oh yeah ignore what I was saying, let's go with @connorjclark !

File: lighthouse-viewer/test/drag-and-drop-test.js (hunk 1 above)
Comment (Remexllee) [reply lighthouse-viewer/test/drag-and-drop-test.js]: do you mean add a comment here?

File: lighthouse-viewer/test/drag-and-drop-test.js (hunk 2 above)
Comment (Remexllee) [reply lighthouse-viewer/test/drag-and-drop-test.js]: will fix it. thanks for suggestion.

File: lighthouse-viewer/test/drag-and-drop-test.js (hunk 1 above)
Comment (patrickhulce) [reply lighthouse-viewer/test/drag-and-drop-test.js]: let's ignore this comment in favor of https://github.com/GoogleChrome/lighthouse/pull/9314#discussion_r299229402

File: lighthouse-viewer/test/drag-and-drop-test.js (hunk 4 above)
Comment (Remexllee) [reply lighthouse-viewer/test/drag-and-drop-test.js]: follow up with https://github.com/GoogleChrome/lighthouse/pull/9314#discussion_r299229402

**Pull request merged**